owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt

```diff
@@ -225,3 +231,42 @@ private fun makeIntent(file: File?, context: Context): Intent {
     sendIntent.putExtra(Intent.ACTION_SEND, true) // Send Action
     return sendIntent
 }
+
+fun Activity.checkPasscodeEnforced(securityEnforced: ISecurityEnforced) {
+    val sharedPreferencesProvider = SharedPreferencesProviderImpl(this)
+
+    val passcodeEnforced = this.resources.getBoolean(R.bool.passcode_enforced)
+    val passcodeConfigured = sharedPreferencesProvider.getBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, false)
+    val patternConfigured = sharedPreferencesProvider.getBoolean(PatternLockActivity.PREFERENCE_SET_PATTERN, false)
+
+    if (passcodeEnforced && !passcodeConfigured && !patternConfigured) {
+
+        val options = arrayOf(getString(R.string.security_enforced_first_option), getString(R.string.security_enforced_second_option))
+        var optionSelected = 0
+
+        AlertDialog.Builder(this).apply {
+            setCancelable(false)
+            setTitle(getString(R.string.security_enforced_title))
+            setSingleChoiceItems(options, optionSelected) { _, which -> optionSelected = which }
+            setPositiveButton(android.R.string.ok) { dialog, _ ->
+                when (optionSelected) {
+                    0 -> securityEnforced.optionLockSelected(LockType.PASSCODE)
+                    1 -> securityEnforced.optionLockSelected(LockType.PATTERN)
+                }
+                dialog.dismiss()
+            }
+        }.show()
+    }
+}
+
+fun Activity.manageOptionLockSelected(type: LockType) {
+    when (type) {
+        LockType.PASSCODE -> startActivity(Intent(this, PassCodeActivity::class.java).apply {
+            action = PassCodeActivity.ACTION_REQUEST_WITH_RESULT
+        })
+        LockType.PATTERN -> startActivity(Intent(this, PatternLockActivity::class.java).apply {
+            action = PatternLockActivity.ACTION_REQUEST_WITH_RESULT
+        })
+    }
+}
```

Comment:
This function is only called from here (indirectly, because `optionLockSelected` calls `manageOptionLockSelected`), so maybe it's not worth it to keep it as an extension. This would imply removing `optionLockSelected` from the interface and its implementations, and also making this a private function here (such as `makeIntent`), but I think it simplifies the code and reduces coupling in some way. Do you agree?

Reply:
In other scenarios I have had problems with dialogs and UI usability, so, I prefer to do it through an Interface.